MFVNeutralino/plugins/Vertexer.cc

```diff
@@ -34,2115 +34,2250 @@
+          for (nv[1] = nv[0] + 1; nv[1] != vertices->end(); ++nv[1]) {
 
-                    int max_missdist_sig_idx = std::max_element(trackrefine_trim_ttks_missdist_sig.begin(), trackrefine_trim_ttks_missdist_sig.end()) - trackrefine_trim_ttks_missdist_sig.begin();
-                    // trimmax only one track with the largest miss distance significance at a time
-                    trackrefine_trimmax_ttks.erase(trackrefine_trimmax_ttks.begin() + max_missdist_sig_idx);
-                    double trackrefine_trimmax_v0x = trackrefine_trimmax_v.position().x() - bsx;
-                    double trackrefine_trimmax_v0y = trackrefine_trimmax_v.position().y() - bsy;
-                    double trackrefine_trimmax_v0r = mag(trackrefine_trimmax_v0x, trackrefine_trimmax_v0y);
+            Measurement1D nv_dist = vertex_dist(*nv[0], *nv[1]);
+            printf("  new vertex dist (2d? %i) %7.3f  sig %7.3f\n", use_2d_vertex_dist, nv_dist.value(), nv_dist.significance());
+          }
+        }
+      }
+    }
+  }
+
+  if (histos_output_beforedzfit){
+    fillCommonOutputHists(vertices, fake_bs_vtx, tt_builder, stepEnum::beforedzfit);
+  }
+  //////////////////////////////////////////////////////////////////////
+  // Drop tracks that "move" the vertex too much by refitting without each track.
+  //////////////////////////////////////////////////////////////////////
+  if (max_nm1_refit_dist3 > 0 || max_nm1_refit_distz > 0 || max_nm1_refit_distz_sig > 0) { 
+    /*
+    auto& tks = nt.tracks();
+    for (int it=0, ite = tks.n(); it < ite; it++){
+        std::cout << " which_pv " << (int) tks.which_pv(it) << std::endl; 
+    }
+    */
+    std::vector<int> refit_count(vertices->size(), 0);
+    int iv = 0;
+    for (v[0] = vertices->begin(); v[0] != vertices->end(); ++v[0], ++iv) {
+      if (max_nm1_refit_count > 0 && refit_count[iv] >= max_nm1_refit_count)
+        continue;
+
+      const track_vec tks = vertex_track_vec(*v[0]);
+      const size_t ntks = tks.size();
+      if (ntks < 3)
+        continue;
+      if (verbose) {
+        printf("doing n-%i refit on vertex at %7.4f %7.4f %7.4f with %lu tracks\n", refit_count[iv] + 1, v[0]->x(), v[0]->y(), v[0]->z(), ntks);
+        for (size_t i = 0; i < ntks; ++i)
+          printf("  refit %lu will drop tk pt %7.4f +- %7.4f eta %7.4f +- %7.4f phi %7.4f +- %7.4f dxy %7.4f +- %7.4f dz %7.4f +- %7.4f\n", i, tks[i]->pt(), tks[i]->ptError(), tks[i]->eta(), tks[i]->etaError(), tks[i]->phi(), tks[i]->phiError(), tks[i]->dxy(), tks[i]->dxyError(), tks[i]->dz(), tks[i]->dzError());
+      }
+
+      std::vector<reco::TransientTrack> ttks(ntks - 1);
+      for (size_t i = 0; i < ntks; ++i) {
+        float tkpt_todrop = tks[i]->pt();
+        float tkphi_todrop = tks[i]->phi();
+        float leptkpt_todrop = -1;
+        if (tks[i].id().id() == 155 || tks[i].id().id() == 156) {
+          leptkpt_todrop = tks[i]->pt();
+        }
```

Comment:
@abbywarden  is this indicating tracks (or lepton tracks) that we want to drop? Are they things that actually get dropped from the vertex? And if so, are they always dropped or only when a certain flag is on? I'm having trouble following it. Maybe it's only for plots? If so, let's add a comment mentioning it.

Reply:
Yes this is only for plots -> to look at the lepton track's pT. This would include all tracks, not only the ones that were dropped. Honestly - it is not needed. Can be deleted or moved?

Reply:
If there's a remote chance it could come in handy again, I would just add a comment about it pertaining only to plots. Otherwise, your call!